### ISSUE TITLE
UI hangs when cluster brought up with vagrant

### ISSUE THREAD

**Issue by jason-riddle** [kubernetes/kubernetes#11119]
By using the following when bringing up the cluster
```bash
export KUBERNETES_PROVIDER=vagrant
./cluster/kube-up.sh
```

and waiting for the cluster to come up, trying to reach the UI at
```bash
https://<master>/ui
```

is unsuccessful since the page just hangs

**Comment by iakshay** [comment]
Same issue on a vagrant setup running on OSX. The css/javascript assets are taking way too much time to load

Kubernetes version - 

```
Client Version: version.Info{Major:"0", Minor:"21+", GitVersion:"v0.21.1-252-g3f2ac7864b6405-dirty", GitCommit:"3f2ac7864b64057970cfd23409630952bc6535fd", GitTreeState:"dirty"}
Server Version: version.Info{Major:"0", Minor:"21+", GitVersion:"v0.21.1-252-g3f2ac7864b6405-dirty", GitCommit:"3f2ac7864b64057970cfd23409630952bc6535fd", GitTreeState:"dirty"}
```

**Comment by thockin** [comment]
@preillyme @bcbroussard

**Comment by alexanderguzhva** [comment]
v1.0.0 (build cd82144), ubuntu 14.04. Works in firefox, no problems. Maybe there was a pause/hangup when tried to open https://<master>/ui for the very first time.

**Comment by metmajer** [comment]
Have the same issues when running a Vagrant cluster on my Mac.

**Comment by jason-riddle** [comment]
Steps to reproduce (For me at least)

    export KUBERNETES_PROVIDER=vagrant
    boot2docker up
    boot2docker ssh sudo /etc/init.d/docker restart  # (Fixes Issue in http://git.io/vmtb6)
    $(boot2docker shellinit)
    make clean
    make quick-release

Then

    ./cluster/kube-down.sh
Bringing down cluster using provider: vagrant
==> minion-1: VM not created. Moving on...
==> master: VM not created. Moving on...
Done

    ./cluster/kube-up.sh
< snipped >

Cluster validation succeeded
Done, listing cluster services:

Kubernetes master is running at https://10.245.1.2
KubeDNS is running at https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-dns
KubeUI is running at https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-ui

    curl -u vagrant:vagrant -k https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-ui/\#/dashboard/
(Hangs...)

This was tested on Google Chrome 43.0.2357.132 and Firefox 37.0.1 running on OSX 10.10.3 with Vagrant 1.7.2 using Kubernetes Tag v1.0.0

**Comment by jweissig** [comment]
Webserver is not returning file contents completely for css/js files for the UI, as in you make a request, and you will only get the first ~12.5K of the file. Retry, and you will get a little more. Weird, I know.

Kubernetes Tag v1.0.0
Latest Version: 1.7.3
Chrome version 43.0.2357.132

After the vagrant cluster is up.

Steps to reproduce at the command line:

```
curl --insecure --user vagrant:vagrant https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-ui/assets/js/base.js
```

File contents will partially download then stall.

Steps to reproduce visually:

Open Chrome developer tools and watch the network tab while viewing:

```
https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-ui/#/dashboard/
```

Hit refresh a few time and watch size column.

**Comment by erictune** [comment]
@preillyme @bcbroussard

**Comment by bcbroussard** [comment]
Thanks for the repro steps @jweissig, I'm seeing similar things. For me the `base.js` file will load, but then `app.js` will stall.

I'm also seeing it when running the last container image `v1` on k8s `1.0.0 RC2` with vagrant. Removing the Kubernetes resource limits does not appear to help either. 

This wasn't happening with the `v0.21.0` release where the web ui was first containerized, so it appears to be something outside of the web ui.

The web ui runs fine on GCE, so it appears to be localized to the vagrant setup.

@timstclair did you see anything like this when testing the initial container? @lavalamp any ideas on what might have changed this?

I'm still looking into it.

**Comment by timstclair** [comment]
I haven't seen anything like that, but I've only tested on GCE.

I'm curious: does the swagger-ui load normally (it's still hosted by the master)? How about other cluster addons (e.g. `/api/v1/proxy/namespaces/kube-system/services/kibana-logging`)? If those both load normally it might be a problem with the way the kube-ui is set up. If other cluster addons have problems, it sounds like it could be a cluster networking issue.

**Comment by lavalamp** [comment]
Since our proxy e2e test is ever so slightly flaky, my money is on some networking trouble.

**Comment by jweissig** [comment]
@timstclair swagger-ui load without issue and show/hide API enpoints (watch requests complete via dev tools too).

```
https://10.245.1.2/swagger-ui/
```

I don't have kibana-logging installed so that is 404'n.

**Comment by bcbroussard** [comment]
@timstclair The issue also exists with `v0.21.0` which is the first release with the containerized version of `kube-ui`

**Comment by laguiz** [comment]
Similar issue here on macos and vagrant :https://github.com/GoogleCloudPlatform/kubernetes/issues/11697 . All endpoints works but not the /ui .

**Comment by jason-riddle** [comment]
Alright, so I've been poking around on this for a few days and I see that the kube-api server takes the following command line option.  

```
--admission-control="NamespaceLifecycle,NamespaceExists,LimitRanger,ResourceQuota"
```

Since kube-ui is being served behind the api server (correct me if I'm wrong), would the ResourceQuota limit how info can be queued at a time, thus causing the only parts of the <code>app.js</code> file to load? Also, could there be a difference in how Saltstack is configuring the vagrant box which does not match the environment that GCE is using? Just some thoughts..

**Comment by lavalamp** [comment]
ResourceQuota doesn't include network traffic at the moment, and definitely wouldn't affect anything but pod creation. That is not the issue. I still think it's some slight (vagrant specific?) network misconfiguration, or a bug in apiserver proxy.

**Comment by c2nes** [comment]
I'm hitting this as well and can provide some additional details. It doesn't appear to be related to the the apiserver proxy from what I can tell. I can reproduce the issue by curling the pod IP directly from the master node, e.g.

```
[vagrant@kubernetes-master ~]$ kubectl --namespace=kube-system describe pod kube-ui | grep IP
IP:				10.246.1.5
[vagrant@kubernetes-master ~]$ curl -so /dev/null 10.246.1.5:8080/assets/js/base.js
...hangs indefinitely more often then not...
```

However, if I issue the same curl command from the minion I can not reproduce the hanging.

I'm running into this with the 1.0.1 release on Vagrant on OS X as well as recent master on Vagrant under Debian. Maybe something funky with Open vSwitch or Vagrant private network config?

**Comment by arun-gupta** [comment]
https://github.com/kubernetes/kubernetes/issues/12692 reports similar issue and was closed. Seems like its a "question" as opposed to "bug" and so need to be asked on stackoverflow. And so posted the question at: http://stackoverflow.com/questions/32226436/kubeui-not-loading-up-with-vagrant

**Comment by derekwaynecarr** [comment]
I will try and take a stab to investigate this more today.

**Comment by derekwaynecarr** [comment]
@rajatchopra @mrunalp - looking to borrow some of your networking expertise.

see https://github.com/kubernetes/kubernetes/issues/11119#issuecomment-126960694

We are having trouble serving the kubernetes-ui through the proxy, it looks like network stalls, tips for debugging further?

**Comment by derekwaynecarr** [comment]
To add some more detail:

For me, the first time I invoke:

```
curl --insecure --user vagrant:vagrant https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-ui/assets/js/base.js
```

I always get back a full response, but I if I then immediately request the resource again in the same terminal window, I get back a partial response.

open a new terminal window, invoke command, get full response.

**Comment by aleks-f** [comment]
Same problem with Vagrant/VirtualBox, k8s 1.0.5, OSX Yosemite 10.5.5. 

Chrome Network Console:
```
GET https://10.245.1.2/api/v1/proxy/namespaces/kube-system/services/kube-ui/assets/js/base.js net::ERR_CONTENT_LENGTH_MISMATCH 10.245.1.2/:54 
app.js:9 Uncaught ReferenceError: angular is not defined
(anonymous function) @ app.js:9
```

**Comment by derekwaynecarr** [comment]
I have a fix for this finally that is pending in the 1.1 stream in this PR:
https://github.com/kubernetes/kubernetes/pull/13986

If you guys can test out and comment that it also fixes your issue, it would be much appreciated and I will then look to backport it to 1.0.x stream.

Apologize for the delay in getting this resolved, but the kube-ui now works nicely with the referenced PR.

**Comment by tsl0922** [comment]
@derekwaynecarr I can confirm #13986 fixes this issue for me, everything works fine, except that there is a failed unit(`docker-storage-setup.service`) on both master and minion, log:

```
[root@kubernetes-minion-1 vagrant]# journalctl -ru docker-storage-setup.service
-- Logs begin at Tue 2015-02-24 15:27:58 UTC, end at Wed 2015-09-16 00:04:44 UTC. --
Sep 15 23:25:00 kubernetes-minion-1 systemd[1]: docker-storage-setup.service failed.
Sep 15 23:25:00 kubernetes-minion-1 systemd[1]: Unit docker-storage-setup.service entered failed state.
Sep 15 23:25:00 kubernetes-minion-1 systemd[1]: Failed to start Docker Storage Setup.
Sep 15 23:25:00 kubernetes-minion-1 systemd[1]: docker-storage-setup.service: main process exited, code=exited, status=1/FAILURE
Sep 15 23:25:00 kubernetes-minion-1 docker-storage-setup[3186]: There is not enough free space in volume group vg_vagrant to create data volume of size MIN_DATA_SIZE=2G.
Sep 15 23:25:00 kubernetes-minion-1 docker-storage-setup[3186]: Logical volume "docker-poolmeta" created
Sep 15 23:24:59 kubernetes-minion-1 docker-storage-setup[3186]: Rounding up size to full physical extent 64.00 MiB
Sep 15 23:24:59 kubernetes-minion-1 systemd[1]: Starting Docker Storage Setup...
```

**Comment by derekwaynecarr** [comment]
@tsl0922 - thanks, can you comment as such on #13986

**Comment by derekwaynecarr** [comment]
@tsl0922 - the docker storage setup is fine because it will default to devicemapper.

**Comment by tsl0922** [comment]
@derekwaynecarr Sure, thanks for your work :+1:

**Comment by derekwaynecarr** [comment]
See #14052 for when the fix will be merged in the next 1.0.x release, in the interim feel free to work from master branch where fix was already merged.

**Comment by arun-gupta** [comment]
Checked out a fresh copy and building it gives:

```
# k8s.io/kubernetes/cmd/hyperkube
/usr/src/go/pkg/tool/linux_amd64/6l: running gcc failed: Cannot allocate memory
!!! Error in /go/src/k8s.io/kubernetes/hack/lib/golang.sh:388
  'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${goldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:388 kube::golang::build_binaries_for_platform(...)
  2: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:523 kube::golang::build_binaries(...)
  3: hack/build-cross.sh:28 main(...)
Exiting with status 1
```

Any idea?

**Comment by tsl0922** [comment]
@arun-gupta Are you build it inside a vm? Try to increase your vm's memory to 3G or higher, this works for me, see #11852

**Comment by arun-gupta** [comment]
@tsl0922 Building it on OSX with 16GB memory.

**Comment by tsl0922** [comment]
@arun-gupta Building kubernetes release on OSX requires docker, If your docker vm is created by the build script, it's memory is 1G(docker-toolbox default) or 2G(boot2docker default), see [the script](https://github.com/kubernetes/kubernetes/blob/master/build/common.sh#L166). 

You may want to create the vm yourself with more memory, and set environment(display it by `boot2docker shellinit` or `docker-machine env kube-dev`) for docker client before make.

**Comment by arun-gupta** [comment]
@tsl0922 how much memory is required to run the build? I can change the script accordingly.

**Comment by tsl0922** [comment]
@arun-gupta I can confirm 1GB is not enough, I built it with 2GB(running `make quick-release`).

**Comment by arun-gupta** [comment]
@tsl0922 Changed Docker Machine memory to 2048 and now getting:

```
Pulling repository gcr.io/google_containers/pause
Pulling repository gcr.io/google_containers/kube-registry-proxy
9b9342c134bb: Pulling image (0.3) from gcr.io/google_containers/kube-registry-proxy, endpoint: https://gcr.io/v1/ 
2c40b0526b63: Error pulling image (0.8.0) from gcr.io/google_containers/pause, endpoint: https://gcr.io/v1/, Get https://storage.googleapis.com/artifacts.google-contain9b9342c134bb: Error pulling image (0.3) from gcr.io/google_containers/kube-registry-proxy, endpoint: https://gcr.io/v1/, Get https://storage.googleapis.com/artifacts.go9b9342c134bb: Error pulling image (0.3) from gcr.io/google_containers/kube-registry-proxy, Get https://storage.googleapis.com/artifacts.google-containers.appspot.com/containers/images/9b9342c134bbda6f57c95c52cafd4ae40333dd6cc0fce4bace9296f957042d22/ancestry: dial tcp: lookup storage.googleapis.com on 93.158.32.20:53: cannot unmarshal Error pulling image (0.8.0) from gcr.io/google_containers/pause, Get https://storage.googleapis.com/artifacts.google-containers.appspot.com/containers/images/2c40b0526b6358710fd09e7b8c022429268cc61703b4777e528ac9d469a07ca1/ancestry: dial tcp: lookup storage.googleapis.com on 93.158.32.20:53: cannot unmarshal DNS message
Error pulling image (0.3) from gcr.io/google_containers/kube-registry-proxy, Get https://storage.googleapis.com/artifacts.google-containers.appspot.com/containers/images/9b9342c134bbda6f57c95c52cafd4ae40333dd6cc0fce4bace9296f957042d22/ancestry: dial tcp: lookup storage.googleapis.com on 93.158.32.20:53: cannot unmarshal DNS message
!!! Error in ./build/../build/common.sh:729
!!! Error in ./build/../build/common.sh:729
  'docker pull "${addon_path}"' exited with status 1
  'docker pull "${addon_path}"' exited with status 1
Call stack:
Call stack:
  1: ./build/../build/common.sh:729 kube::release::write_addon_docker_images_for_server(...)
  1: ./build/../build/common.sh:729 kube::release::write_addon_docker_images_for_server(...)
  2: ./build/../build/common.sh:651 kube::release::package_server_tarballs(...)
  2: ./build/../build/common.sh:651 kube::release::package_server_tarballs(...)
  3: ./build/../build/common.sh:586 kube::release::package_tarballs(...)
  3: ./build/../build/common.sh:586 kube::release::package_tarballs(...)
  4: ./build/release.sh:40 main(...)
  4: ./build/release.sh:40 main(...)
Exiting with status 1
Exiting with status 1
!!! [0917 17:07:49] unable to pull or write addon image
!!! [0917 17:07:49] previous tarball phase failed
!!! Error in ./build/release.sh:40
  'return 1' exited with status 1
Call stack:
  1: ./build/release.sh:40 main(...)
Exiting with status 1
```

**Comment by aviau** [comment]
I am still getting this exact error with v1.2.0-alpha. Does that make sense?

**Comment by derekwaynecarr** [comment]
There are multiple unrelated problems being discussed in this issue.

Can the discussion move to a new dedicated issue for each particular topic
(I.e building code, etc)?

**Comment by aviau** [comment]
In my case, it is the same symptoms as the original issue.

**Comment by derekwaynecarr** [comment]
I can't check easily right now, but can you check if flannel is running in your
environment?

**Comment by rahulsrivastava71** [comment]
Hi team , i am running 

./hyperkube kube-apiserver --etcd-servers=http://localhost:2379 --service-cluster-ip-range=10.0.0.0/16 --bind-address=0.0.0.0 --insecure-bind-address=0.0.0.0 --admission-control= --insecure-port 12345
cammand and facinf issue of OOO as my program gets stuck at 


# ./hyperkube kube-apiserver --etcd-servers=http://localhost:2379 --service-cluster-ip-range=10.0.0.0/16 --bind-address=0.0.0.0 --insecure-bind-address=0.0.0.0 --admission-control= --insecure-port 12345
Flag --insecure-bind-address has been deprecated, This flag will be removed in a future version.
Flag --admission-control has been deprecated, Use --enable-admission-plugins or --disable-admission-plugins instead. Will be removed in a future version.
Flag --insecure-port has been deprecated, This flag will be removed in a future version.
I0913 05:12:45.447088    3176 server.go:703] external host was not specified, using 10.98.4.112
W0913 05:12:45.447311    3176 authentication.go:378] AnonymousAuth is not allowed with the AlwaysAllow authorizer. Resetting AnonymousAuth to false. You should use a different authorizer
I0913 05:12:45.447894    3176 server.go:145] Version: v1.11.0
I0913 05:12:46.343968    3176 master.go:234] Using reconciler: lease
W0913 05:12:47.373485    3176 genericapiserver.go:319] Skipping API batch/v2alpha1 because it has no resources.
W0913 05:12:47.811657    3176 genericapiserver.go:319] Skipping API rbac.authorization.k8s.io/v1alpha1 because it has no resources.
W0913 05:12:47.818034    3176 genericapiserver.go:319] Skipping API scheduling.k8s.io/v1alpha1 because it has no resources.
W0913 05:12:47.896044    3176 genericapiserver.go:319] Skipping API storage.k8s.io/v1alpha1 because it has no resources.
W0913 05:12:48.617806    3176 genericapiserver.go:319] Skipping API admissionregistration.k8s.io/v1alpha1 because it has no resources.
[restful] 2018/09/13 05:12:48 log.go:33: [restful/swagger] listing is available at https://10.98.4.112:6443/swaggerapi
[restful] 2018/09/13 05:12:48 log.go:33: [restful/swagger] https://10.98.4.112:6443/swaggerui/ is mapped to folder /swagger-ui/

when i check the logs it shows that it got killed due to OOO.
i am using kubernetes 1.11.0 version and my memory is 512mb (virtual box)
